backend/python-functions/helpers.py

```diff
@@ -0,0 +1,68 @@
+# adapted from https://codereview.stackexchange.com/questions/217303/decorate-a-python-function-to-work-as-a-google-cloud-function
+import functools
+from flask import json
+import traceback
+
+import jsonschema
+
+
+def cloudfunction(in_schema=None, out_schema=None):
+    """
+    :param in_schema: the schema for the input, or a falsy value if there is no input
+    :param out_schema: the schema for the output, or a falsy value if there is no output
+    :return: the cloudfunction wrapped function
+    """
+    # Both schemas must be valid according to jsonschema draft 7, if they are provided.
+    if in_schema:
+        jsonschema.Draft7Validator.check_schema(in_schema)
+    if out_schema:
+        jsonschema.Draft7Validator.check_schema(out_schema)
+
+    def cloudfunction_decorator(f):
+        """ Wraps a function with one argument, a json object that it expects to be sent with the request.
+        It modifies it by setting CORS headers and responding to OPTIONS requests with `Allow-Origin *`
+
+        :param f: A function that takes a `request` and returns a json-serializable object
+        :return: a function that accepts one argument, a Flask request, and calls f with the modifications listed
+        """
+
+        @functools.wraps(f)
+        def wrapped(request):
+            if request.method == 'OPTIONS':
+                return cors_options()
+
+            # If it's not a CORS OPTIONS request, still include the base header.
+            headers = {'Access-Control-Allow-Origin': '*'}
+
+            try:
+                if in_schema:
+                    request_json = request.get_json()
+                    jsonschema.validate(request_json, in_schema)
```

Comment:
Do these validate calls automatically return 400/BAD_REQUEST on failure?

Reply:
no, they just raise an exception that gets caught on L50 below; then we return a 500 response on L51. It's very generic, but we could tease that out to provide more meaningful error codes if you think that's worthwhile now

Reply:
That seems sufficient for now.